executor/simpletest/simple_test.go

```diff
@@ -880,6 +884,96 @@ func TestUser(t *testing.T) {
+	createAndCheck(tk, "CREATE USER 'u4'@'localhost' IDENTIFIED BY 'password' PASSWORD_LOCK_TIME 3;", "{\"Password_locking\": {\"failed_login_attempts\": 0, \"password_lock_time_days\": 3}}", "u4")
+	createAndCheck(tk, "CREATE USER 'u5'@'localhost' IDENTIFIED BY 'password' PASSWORD_LOCK_TIME UNBOUNDED;", "{\"Password_locking\": {\"failed_login_attempts\": 0, \"password_lock_time_days\": -1}}", "u5")
+
+	alterAndCheck(t, tk, "ALTER USER 'u1'@'localhost' FAILED_LOGIN_ATTEMPTS 4 PASSWORD_LOCK_TIME 6;", "u1", 4, 6)
+	alterAndCheck(t, tk, "ALTER USER 'u2'@'localhost' FAILED_LOGIN_ATTEMPTS 4 PASSWORD_LOCK_TIME UNBOUNDED;", "u2", 4, -1)
+	alterAndCheck(t, tk, "ALTER USER 'u3'@'localhost' PASSWORD_LOCK_TIME 6;", "u3", 3, 6)
+	alterAndCheck(t, tk, "ALTER USER 'u4'@'localhost' FAILED_LOGIN_ATTEMPTS 4;", "u4", 4, 3)
+	alterAndCheck(t, tk, "ALTER USER 'u4'@'localhost' PASSWORD_LOCK_TIME UNBOUNDED;", "u4", 4, -1)
+
+	createAndCheck(tk, "CREATE USER 'u6'@'localhost' IDENTIFIED BY '' FAILED_LOGIN_ATTEMPTS 3 PASSWORD_LOCK_TIME 3;", "{\"Password_locking\": {\"failed_login_attempts\": 3, \"password_lock_time_days\": 3}}", "u6")
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u6", Hostname: "localhost"}, encodePassword("password"), nil))
+	checkAuthUser(t, tk, "u6", 1)
+	require.NoError(t, tk.Session().Auth(&auth.UserIdentity{Username: "u6", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+}
+
+func encodePassword(password string) []byte {
+	pwd := auth.EncodePassword(password)
+	hpwd, _ := auth.DecodePassword(pwd)
+	return hpwd
+}
+
+func createAndCheck(tk *testkit.TestKit, sql, rsJSON, user string) {
+	tk.MustExec(sql)
+	sql = selectSQL(user)
+	tk.MustQuery(sql).Check(testkit.Rows(rsJSON))
+}
+
+func alterAndCheck(t *testing.T, tk *testkit.TestKit, sql string, user string, failedLoginAttempts, passwordLockTimeDays int64) {
+	tk.MustExec(sql)
+	userAttributesSQL := selectSQL(user)
+	resBuff := bytes.NewBufferString("")
+	for _, row := range tk.MustQuery(userAttributesSQL).Rows() {
+		_, _ = fmt.Fprintf(resBuff, "%s\n", row)
+	}
+	checkUser(t, resBuff.String(), failedLoginAttempts, passwordLockTimeDays)
+}
+
+func checkUser(t *testing.T, rs string, failedLoginAttempts int64, passwordLockTimeDays int64) {
+	var ua []userAttributes
+	if err := json.Unmarshal([]byte(rs), &ua); err == nil {
+		require.True(t, ua[0].PasswordLocking.FailedLoginAttempts == failedLoginAttempts)
+		require.True(t, ua[0].PasswordLocking.PasswordLockTimeDays == passwordLockTimeDays)
+	} else {
+		fmt.Println(err)
```

Comment:
Please return the error, receive the error in the `TestXXX` function, and use `require.NoError(t, err)` to make sure that no error happens. Like
```go
err = alterAndCheck(xxx)
require.NoError(t, err)
```